kore/src/Kore/Step/Simplification/Exists.hs

```diff
@@ -196,194 +195,199 @@ makeEvaluate
-            (result, _proof) <-
-                ExpandedPattern.simplify
-                    tools
-                    substitutionSimplifier
-                    simplifier
-                    axiomIdToSimplifier
-                    substitutedPat
-            return (result , SimplificationProof)
+    original
+  = fmap withProof $ gather $ do
+    normalized <- normalize original
+    let Predicated { substitution = normalizedSubstitution } = normalized
+    case splitSubstitution variable normalizedSubstitution of
+        (Left boundTerm, freeSubstitution) ->
+            makeEvaluateBoundLeft
+                tools
+                substitutionSimplifier
+                simplifier
+                axiomIdToSimplifier
+                variable
+                boundTerm
+                normalized { substitution = freeSubstitution }
+        (Right boundSubstitution, freeSubstitution) ->
+            makeEvaluateBoundRight
+                variable
+                freeSubstitution
+                normalized { substitution = boundSubstitution }
   where
-    (boundSubstitution, freeSubstitution) =
-        splitSubstitutionByVariable variable $ Substitution.toMap substitution
-
-{- | Existentially quantify a variable over an 'ExpandedPattern'.
-
-The input is a pattern of the form
-
-@
-φ ∧ P ∧ S
-@
-
-where @P@ is a predicate and @S@ is a substitution, having the form
-
-@
-S = ∧ᵢ xᵢ = tᵢ
-@
-
-where @xᵢ@ is a variable. The quantified variable @y@ does not occur on the
-left-hand side of any conjunct in @S@, but this is not checked. The quantifier
-will be lowered into the pattern as far as possible.
+    withProof a = (a, SimplificationProof)
+    normalize =
+        Substitution.normalize
+            tools
+            substitutionSimplifier
+            simplifier
+            axiomIdToSimplifier
 
-The predicate @P@ is split into a bound part and a free part,
+{- | Existentially quantify a variable in the given 'ExpandedPattern'.
 
-@
-P = freeP ∧ boundP
-@
+The variable was found on the left-hand side of a substitution and the given
+term will be substituted everywhere. The variable may occur anywhere in the
+'term' or 'predicate' of the 'ExpandedPattern', but not in the
+'substitution'. The final result will not contain the quantified variable and
+thus the quantifier will actually be omitted.
 
-such that @freeP@ does not contain @y@, but @boundP@ (if it exists) does contain
-@y@. Likewise the substitution is split into a bound part and a free part,
-
-@
-S = freeS ∧ boundS
-@
-
-such that @y@ does not occur on the right-hand side of any conjunct in @freeS@,
-but @y@ occurs on the right-hand side of /every/ conjunct in @boundS@.
-
-If @y@ occurs in @φ@, the result is
-
-@
-(∃ y. φ ∧ boundP ∧ boundS) ∧ freeP ∧ freeS
-@
-
-otherwise (if @y@ does not occur in @φ@) the result is
-
-@
-φ ∧ (freeP ∧ (∃ y. boundP ∧ boundS)) ∧ freeS.
-@
+See also: 'quantifyExpandedPattern'
```

Comment:
This function relies on `boundTerm` not containing `variable` as a free variable, it may be worth adding that constraint here (another way would be to use a normalized substitution, but that may not be worth the effort).

Reply:
I documented this fact and added a check.